benchbuild/experiment.py

```diff
@@ -24,23 +24,36 @@ class HelloExperiment(Experiment):
 ```
 
 """
+from __future__ import annotations
+
 import collections
 import copy
 import uuid
 from abc import abstractmethod
+from typing import TYPE_CHECKING
 
 import attr
 
 from benchbuild.settings import CFG
+from benchbuild.utils import validators
 from benchbuild.utils.actions import (Any, Clean, CleanExtra, Compile,
                                       Containerize, Echo, MakeBuildDir,
                                       RequireAll, Run)
 
+if TYPE_CHECKING:
+    from benchbuild.project import Project
+    from benchbuild.utils.actions import Step
+
+    from typing import Dict, Iterable, List, Type, TypeVar
+
+    Step_co = TypeVar('Step_co', bound=Step)
```

Comment:
C0103 (invalid-name): Class name "Step_co" doesn't conform to '[A-Z_][a-zA-Z0-9]+$' pattern  <!-- Automatically posted by Lintly -->